src/main/java/com/google/step/datamanager/FollowManagerDatastore.java

```diff
@@ -119,6 +120,22 @@ private void unfollowSomething(long followerId, long fieldId, String fieldName)
   }
 
   @Override
+  public void updateFollowedTagIds(long followerId, List<Long> tagIds) {
+    List<Long> tagsFollowed = getFollowedTagIds(followerId);
+    HashSet<Long> newTags = new HashSet<>(tagIds);
+    for (long id : tagsFollowed) {
+      if (newTags.contains(id)) {
+        newTags.remove(id);
+      } else {
+        unfollowTag(followerId, id);
+      }
+    }
```

Comment:
I think `getFollowedTagIds` should return a Set<> as well. Otherwise, your logic may have a bug when there are duplicates in `tagsFollowed` since it'll end up unfollowing them.

Reply:
should the getFollowedTagIds return Set or I convert the return value to Set?

Reply:
It should return Set, returning a List when elements are supposed to be unique can be confusing.

Reply:
ok I see